zebra-consensus/src/primitives/ed25519.rs

```diff
@@ -120,36 +119,41 @@ impl Verifier {
 
     /// Flush the batch using a thread pool, and return the result via the channel.
     /// This function returns a future that becomes ready when the batch is completed.
-    fn flush_spawning(batch: BatchVerifier, tx: Sender) -> impl Future<Output = ()> {
+    async fn flush_spawning(batch: BatchVerifier, tx: Sender) {
+        let (rsp_tx, rsp_rx) = tokio::sync::oneshot::channel();
+
         // Correctness: Do CPU-intensive work on a dedicated thread, to avoid blocking other futures.
-        tokio::task::spawn_blocking(|| {
+        rayon::spawn_fifo(|| {
             // TODO:
             // - spawn batches so rayon executes them in FIFO order
             //   possible implementation: return a closure in a Future,
             //   then run it using scope_fifo() in the worker task,
             //   limiting the number of concurrent batches to the number of rayon threads
-            rayon::scope_fifo(|s| s.spawn_fifo(|_s| Self::verify(batch, tx)))
-        })
-        .map(|join_result| join_result.expect("panic in ed25519 batch verifier"))
+            Self::verify(batch, tx);
+            let _ = rsp_tx.send(());
+        });
+
+        rsp_rx.await.expect("sender should not be dropped")
```

Comment:
This could panic during shutdown depending on the order of sender/receiver drops.
(This applies to all repetitions of this code in this PR.)

Reply:
I'm not sure if you meant to resolve this?

Users have opened bug reports for panics on shutdown, so I'd like to fix this.

I think it would be better to either:
* ignore any errors, optionally logging them with a note "Is Zebra shutting down?"
* if we need to return something, return the closest Result::Err variant

We can't return `Ok(())`, because technically that could make verification of an invalid proof or signature pass, and get committed to the finalized state.

Reply:
Fixed in https://github.com/ZcashFoundation/zebra/pull/6887/commits/e4fcc93ed2b85adfcf6eb9fcdc912afd5bfd2273, updated to use `BoxError` in https://github.com/ZcashFoundation/zebra/pull/6887/commits/8c6fb80e8af4445bc096255299deba35dc62768a, simplified in https://github.com/ZcashFoundation/zebra/pull/6887/commits/08649438e91f6dce7bfa258add4b8911280f7e22 and https://github.com/ZcashFoundation/zebra/pull/6887/commits/08649438e91f6dce7bfa258add4b8911280f7e22.

> I'm not sure if you meant to resolve this?

No, probably a mis-click.

> if we need to return something, return the closest Result::Err variant

I went with `BoxError` to avoid repetitive or confusing logs.